components/services/brave_wallet/public/cpp/BUILD.gn

```diff
@@ -0,0 +1,29 @@
+# Copyright (c) 2024 The Brave Authors. All rights reserved.
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at https://mozilla.org/MPL/2.0/.
+
+static_library("cpp") {
```

Comment:
component

Reply:
- We don't need that many `component()`. There should be not more than one.
- We don't need that one `component()` all. Just `source_set()` should be enough as amount of code is relatively small in this service, we have only one consumer and this consumer is not a `component()` itself

Reply:
Also let's not invent the wheel and just reuse architecture and layout of these services:
`/src/components/services/patch/`
`/src/components/services/unzip/`
`/src/services/data_decoder/`

Reply:
I'd tried to keep data decoder layout where services/../public/cpp folder contains service client implementation. 
But i don't like how data_decoder service manages process launch model since it mixes external static delegate setup with internal buildflags.
I don't think it is a good idea to setup process launch model via static delegate since we would also need to ensure that it is initialized at the moment of usage.
So i've added buildflags in brave_wallet_utils_service to decide what process model to use.
Also data_decoder provides a set of static methods to use, but 
1) It always starts a new process on every such static call which could be surprising. This affects performance.
2) Number of data_decoder parsing formats is limited, but in our case BraveWalletUtilsService could be extended with other coins format, rust libs, etc..

To overcome this issues, BraveWalletUtilsService acts as a factory for sub-component interfaces:
If client code wants to use separate process, it creates BraveWalletUtilsInstance manually.
If client code wants to reuse general process, it uses GetInstance() singletone.
User should explicitly request which sub-component to use. For ex. call CreateZCashDecoder.